src/components/map/map.tsx

```diff
@@ -0,0 +1,234 @@
+import React, { useRef, useEffect, useState, useCallback } from 'react';
+import mapboxgl from 'mapbox-gl';
+import style from './map.module.css';
+import 'mapbox-gl/dist/mapbox-gl.css';
+import { mapboxData } from '@atb/modules/org-data';
+import { isFeaturePoint, isStopPlace } from './utils';
+import { ColorIcon } from '@atb/assets/color-icon';
+import Button from '../button/button';
+import { MonoIcon } from '@atb/assets/mono-icon';
+import { ComponentText, useTranslation } from '@atb/translations';
+
+export type LngLatPosition = [number, number];
+export type MapProps = {
+  position?: LngLatPosition;
+  layer?: string;
+  onSelectStopPlace: (id: string) => void;
+};
+
+type Marker = HTMLElement &
+  React.DetailedReactHTMLElement<
+    {
+      className: string;
+    },
+    HTMLElement
+  >;
+
+const INTERACTIVE_LAYERS = [
+  'airport.nsr.api',
+  'boat.nsr.api',
+  'bus.nsr.api',
+  'bus.tram.nsr.api',
+  'bussterminal.nsr.api',
+  'ferjekai.nsr.api',
+  'metro.nsr.api',
+  'metro.tram.nsr.api',
+  'railway.nsr.api',
+  'tram.nsr.api',
+];
+
+export function Map({
+  position = defaultPosition,
+  layer,
+  onSelectStopPlace,
+}: MapProps) {
+  const mapContainer = useRef<HTMLDivElement>(null);
+  const map = useRef<mapboxgl.Map>();
+  const { t } = useTranslation();
+
+  useEffect(() => {
+    if (!mapContainer.current) return;
+    map.current = new mapboxgl.Map({
+      container: mapContainer.current,
+      accessToken: mapboxData.accessToken,
+      style: mapboxData.style,
+      center: position,
+      zoom: 13,
+    });
+
+    return () => map.current?.remove();
+  }, []); // eslint-disable-line react-hooks/exhaustive-deps
+
+  const { centerMap } = useMapInteractions(map, onSelectStopPlace);
+  const { mapWrapperRef, toggleFullscreen } = useFullscreenMap(map);
+  useMapPin(map, position, layer);
+
+  useEffect(() => {
+    if (map.current) {
+      map.current.flyTo({
+        center: position,
+        zoom: 15,
+        speed: 2,
+      });
+    }
+  }, [position]);
+
+  return (
+    <div className={style.map}>
+      <Button
+        className={style.fullscreenButton}
+        title={t(ComponentText.Map.map.openFullscreenButton)}
+        icon={{ left: <MonoIcon src="map/Map.svg" /> }}
```

Comment:
I think we should have used the other Icon component from Webshop. But I can fix that in another PR

Reply:
(meaning the `@atb/component/icon` one)